parametric_plasma_source/plasma.py

```diff
@@ -0,0 +1,380 @@
+
+    @ion_temperature_pedistal.setter
+    def ion_temperature_pedistal(self, ion_temperature_pedistal):
+        if ion_temperature_pedistal < 0:
+            raise ValueError('ion_temperature_pedistal is out of range')
+        else:
+            self._ion_temperature_pedistal = ion_temperature_pedistal
+
+    @property
+    def ion_density_origin(self):
+        return self._ion_density_origin
+
+    @ion_density_origin.setter
+    def ion_density_origin(self, ion_density_origin):
+        if ion_density_origin < 0:
+            raise ValueError('ion_density_origin is out of range')
+        else:
+            self._ion_density_origin = ion_density_origin
+
+    @property
+    def ion_density_seperatrix(self):
+        return self._ion_density_seperatrix
+
+    @ion_density_seperatrix.setter
+    def ion_density_seperatrix(self, ion_density_seperatrix):
+        if ion_density_seperatrix < 0:
+            raise ValueError('ion_density_seperatrix is out of range')
+        else:
+            self._ion_density_seperatrix = ion_density_seperatrix
+
+    @property
+    def triangularity(self):
+        return self._triangularity
+
+    @triangularity.setter
+    def triangularity(self, triangularity):
+        if triangularity > 2000 or triangularity < -2000:
+            raise ValueError('triangularity is out of range')
+        else:
+            self._triangularity = triangularity
+
+    @property
+    def single_null(self):
+        return self._single_null
+
+    @single_null.setter
+    def single_null(self, single_null):
+        if type(single_null) != bool :
+            raise ValueError('single_null must be True or False')
+        else:
+            self._single_null = single_null
+
+    @property
+    def minor_radius(self):
+        return self._minor_radius
+
+    @minor_radius.setter
+    def minor_radius(self, minor_radius):
+        self._minor_radius = minor_radius
+
+    @property
+    def major_radius(self):
+        return self._major_radius
+
+    @major_radius.setter
+    def major_radius(self, major_radius):
+        if major_radius > 2000 or major_radius < 1:
+            raise ValueError('major_radius is out of range')
+        else:
+            self._major_radius = major_radius
+
+    @property
+    def elongation(self):
+        return self._elongation
+
+    @elongation.setter
+    def elongation(self, elongation):
+        if elongation > 4 or elongation < 0:
+            raise ValueError('elongation is out of range')
+        else:
+            self._elongation = elongation
+
+    @property
+    def ion_density_pedistal(self):
+        return self._ion_density_pedistal
+
+    @ion_density_pedistal.setter
+    def ion_density_pedistal(self, ion_density_pedistal):
+        if ion_density_pedistal > 10e22 or ion_density_pedistal < 1e4:
+            raise ValueError('ion_density_pedistal is out of range')
+        else:
+            self._ion_density_pedistal = ion_density_pedistal
+
+    def replace_variable_value(
```

Comment:
what does this do?

Reply:
This is an artefact of the makefile approach to get the variable values into the shared object that gets built. It will go away when the serialisation approach is ready.